vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1278,17 +1278,22 @@ def apply_grammar_bitmask(
             cumulative_index += 1 + num_spec_tokens
         grammar_bitmask = sorted_bitmask
 
+        # If the grammar bitmask and the logits have the same shape
+        # we don't need to pass indices to the kernel,
+        # since the bitmask is already aligned with the logits.
+        skip_out_indices = grammar_bitmask.shape[0] == logits.shape[0]
+
         # Serialization of np.ndarray is much more efficient than a tensor,
         # so we receive it in that format.
-        grammar_bitmask = torch.from_numpy(grammar_bitmask)
+        grammar_bitmask = torch.from_numpy(grammar_bitmask).contiguous()
```

Comment:
This is just for sanity, could someone confirm that the above numpy-indexing logic with `torch.from_numpy` will always create a contiguous array and not a row-wise view?

Reply:
afaik torch.from_numpy creates a view, so as long as the original numpy array is contiguous, it should be ok?